### ISSUE TITLE
barcode scan crash

### ISSUE THREAD

**Issue by mvayngrib** [Anyline/anyline-ocr-react-native-module#8]
barcodeFormat is nil and `dictResult` is not happy!

![image](https://cloud.githubusercontent.com/assets/83948/22129621/1bae389e-de75-11e6-872b-25fae51505fe.png)

in case it helps, this is the me printing `scanResult`:
```
(lldb) po scanResult
@

ANSI 636001070002DL00410392ZN04330047DLDCANONE  
DCBNONE        
DCDNONE 
DBA08312013
DCSMichael                                 
DACM                                       
DADMotorist                                
DBD08312013
DBB08312013
DBC1
DAYBRO
DAU064 in
DAG2345 ANYWHERE STREET               
DAIYOUR CITY           
DAJNY
DAK123450000  
DAQNONE                     
DCFNONE                     
DCGUSA
DDEN
DDFN
DDGN

ZNZNAMDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5
```

one of the characters isn't showing up, here's another screenshot:

![image](https://cloud.githubusercontent.com/assets/83948/22129659/63e28606-de75-11e6-90b4-c570870feafe.png)

**Comment by danielalbertini** [comment]
Thx for the bug report. Indeed at the moment we are returning nil when the barcode format is unknown.
We will fix that in the upcoming SDK version.
Additionally we will make the switch to the following delegate call as this one is deprecated since 3.4 (that's why we missed that one in testing).
`- (void)anylineBarcodeModuleView:(AnylineBarcodeModuleView *)anylineBarcodeModuleView
               didFindScanResult:(NSString *)scanResult
               withBarcodeFormat:(ALBarcodeFormat)barcodeFormat
                         atImage:(UIImage *)image;`

**Comment by mvayngrib** [comment]
@danielalbertini yea! I saw it was deprecated and tried changing it to the newer method and I guess I was indeed scanning a bad barcode. Glad to hear it'll be fixed soon

**Comment by JonasLaux** [comment]
Commit [ecfa033](https://github.com/Anyline/anyline-ocr-react-native-module/pull/10/commits/ecfa03334dc306c47331a2c7e488aaf9198d16ef) closes #8
